pkg/node/node.go

```diff
@@ -96,46 +113,17 @@ func New() *Node {
 	csi.RegisterIdentityServer(node.Server, node)
 	csi.RegisterNodeServer(node.Server, node)
 
+	// initialize node communication service
+	go node_service.ListenAndServe(envServicePort)
+
 	return node
 }
 
 // NodeGetInfo returns info about the node
 func (node *Node) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
-	initiatorName, err := readInitiatorName()
-	if err != nil {
-		return nil, status.Error(codes.FailedPrecondition, err.Error())
-	}
-
-	topology := map[string]string{}
-	sasAddresses, err := storage.GetSASInitiators()
-	if err != nil {
-		klog.Warningf("Error while searching for FC HBA Addresses: %s", err)
-	}
-	fcAddresses, err := storage.GetFCInitiators()
-	if err != nil {
-		klog.Warningf("Error while searching for FC HBA Addresses: %s", err)
-	}
-
-	for i, sasAddr := range sasAddresses {
-		//maximum value length 63 chars
-		topoKey := fmt.Sprintf("%s/%s-%d", common.TopologyInitiatorPrefix, common.TopologySASInitiatorLabel, i)
-		topology[topoKey] = sasAddr
-	}
-
-	for i, fcAddr := range fcAddresses {
-		topoKey := fmt.Sprintf("%s/%s-%d", common.TopologyInitiatorPrefix, common.TopologyFCInitiatorLabel, i)
-		topology[topoKey] = fcAddr
-	}
-
-	topology[common.TopologyNodeIDKey] = common.GetTopologyCompliantNodeID(initiatorName)
-
-	klog.Infof("Node Accessible Topology: %v", topology)
 	return &csi.NodeGetInfoResponse{
-		NodeId:            initiatorName,
+		NodeId:            node.nodeIP,
```

Comment:
Could this be a node name instead of IP?  Names tend to be more recognizable and durable; IPs can change if using DHCP, etc.

Reply:
Moving forward with the IP for now since kubernetes makes pod IP easily available to us. If the node reboots and gets a new IP, that will be reflected in the new instance of the pod.